app/views/results_entities.php

```diff
@@ -37,17 +43,8 @@
         }
     }
 
-    if (isset($tmx_target[$entity])) {
-        // nbsp highlight
-        $target_string = str_replace(' ', '<span class="highlight-gray"> </span>', $tmx_target[$entity]);
-    } else {
-        $target_string = '';
-    }
-
-    // Escape strings for HTML Display
-    $source_string  = Utils::secureText($tmx_source[$entity]);
-    $target_string  = Utils::secureText($target_string);
-    $target_string2 = Utils::secureText($target_string2);
+    // nbsp highlight only after strings have been escaped
+    $target_string = str_replace(' ', '<span class="highlight-gray"> </span>', $target_string);
```

Comment:
Same note about the comment, I would move this one to line 30 where you define `$bz_target_string = $target_string`